### PR TITLE
cleanup: fixes cleanup devices.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
- Implement cleanup of orhan device update, seems there were some implementation in the past, were update-transaction had no devices collections , and the devices was linked to update transaction via dispatcher-records. This PR cleanup that update-transaction before cleaning up the main devices with transactions.
- Remove the delete of commit, that will be implemented in its own module. 
FIXES: https://issues.redhat.com/browse/THEEDGE-3515

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by adeliaferreira — DISMISSED** [review]
lgtm

File: cmd/cleanup/cleanupdevices/cleanupdevices.go [hunk 1]
```diff
@@ -259,27 +262,113 @@ func CleanUpDevice(s3Client *files.S3Client, candidateDevice *CandidateDevice) e
...
+		Scan(&candidateDevices).Error
+
+	if err != nil {
+		log.WithFields(log.Fields{"error": err.Error()}).Error("error occurred when collecting orphan devices updates candidates")
+	}
+
+	return candidateDevices, err
+}
+
+func CleanupOrphanDevicesUpdates(s3Client *files.S3Client, gormDB *gorm.DB) error {
```
Comment (mgold1234) [inline cmd/cleanup/cleanupdevices/cleanupdevices.go:294]: can you add description here as well?

File: cmd/cleanup/cleanupdevices/cleanupdevices.go [hunk 2]
```diff
@@ -259,27 +262,113 @@ func CleanUpDevice(s3Client *files.S3Client, candidateDevice *CandidateDevice) e
...
+
+	return nil
+}
+
 // GetCandidateDevices returns the data set of devices cleanup candidates
 func GetCandidateDevices(gormDB *gorm.DB) ([]CandidateDevice, error) {
 	var candidateDevices []CandidateDevice
 
-	if err := gormDB.Debug().Table("devices").
+	err := gormDB.Debug().Table("devices").
```
Comment (mgold1234) [inline cmd/cleanup/cleanupdevices/cleanupdevices.go:352]: is this row necessary?

Comment (ldjebran) [reply cmd/cleanup/cleanupdevices/cleanupdevices.go:352]: yes we need this row  :rocket:

File: cmd/cleanup/cleanupdevices/cleanupdevices.go (hunk 1 above)
Comment (ldjebran) [reply cmd/cleanup/cleanupdevices/cleanupdevices.go:297]: added

**Review by adeliaferreira — APPROVED** [review]
Lgtm

File: cmd/cleanup/cleanupdevices/cleanupdevices.go (hunk 2 above)
Comment (ldjebran) [reply cmd/cleanup/cleanupdevices/cleanupdevices.go:355]: +, I am resolving a bug , in case of error it was not returned

**Review by mgold1234 — APPROVED** [review]
/lgtm

File: cmd/cleanup/cleanupdevices/cleanupdevices.go (hunk 2 above)
Comment (mgold1234) [reply cmd/cleanup/cleanupdevices/cleanupdevices.go:355]: great

File: cmd/cleanup/cleanupdevices/cleanupdevices.go (hunk 1 above)
Comment (mgold1234) [reply cmd/cleanup/cleanupdevices/cleanupdevices.go:297]: thanks

**Pull request merged**